src/app/components/news/news.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit, Inject, Input } from '@angular/core';
+import { NewsService, ActualNewsService } from '../../services/news.service';
+import { News } from '../../models/News';
+
+@Component({
+  selector: 'app-news',
+  templateUrl: './news.component.html',
+  styleUrls: ['./news.component.scss'],
+  // Here we choose which news service is used
+  providers: [{ provide: 'NewsService', useClass: ActualNewsService }]
+})
+export class NewsComponent implements OnInit {
+  news: News[];
+
+  constructor(@Inject('NewsService') newsService: NewsService) {
+    newsService.getNews().subscribe(
```

Comment:
I think in angular, the constructor is generally used for dependency injection and not much else, with other initialisation logic occurring in in the [onInit](https://stackoverflow.com/questions/35763730/difference-between-constructor-and-ngoninit/35763811#35763811) life cycle hook

Reply:
Makes sense, especially when react and svelte do the same thing (put logic inside init component hook).